modules/pricer/src/main/java/com/opengamma/strata/pricer/credit/cds/IsdaCompliantDiscountCurveCalibrator.java

```diff
@@ -0,0 +1,454 @@
+import com.opengamma.strata.math.impl.matrix.CommonsMatrixAlgebra;
+import com.opengamma.strata.math.impl.matrix.MatrixAlgebra;
+import com.opengamma.strata.math.impl.rootfinding.BracketRoot;
+import com.opengamma.strata.math.impl.rootfinding.NewtonRaphsonSingleRootFinder;
+
+/**
+ * ISDA compliant discount curve calibrator.
+ * <p>
+ * A single discounting curve is calibrated for a specified currency.
+ * <p>
+ * The curve is defined using two or more {@linkplain CurveNode nodes}.
+ * Each node primarily defines enough information to produce a reference trade.
+ * Calibration involves pricing, and re-pricing, these trades to find the best fit using a root finder.
+ * <p>
+ * Once calibrated, the curves are then available for use.
+ * Each node in the curve definition becomes a parameter in the matching output curve.
+ */
+public final class IsdaCompliantDiscountCurveCalibrator {
+
+  /**
+   * Default implementation.
+   */
+  public static final IsdaCompliantDiscountCurveCalibrator DEFAULT = new IsdaCompliantDiscountCurveCalibrator();
+
+  /**
+   * The matrix algebra used for matrix inversion.
+   */
+  private static final MatrixAlgebra MATRIX_ALGEBRA = new CommonsMatrixAlgebra();
+  /**
+   * The root bracket finder.
+   */
+  private static final BracketRoot BRACKETER = new BracketRoot();
+  /**
+   * The root finder.
+   */
+  private static final NewtonRaphsonSingleRootFinder ROOT_FINDER = new NewtonRaphsonSingleRootFinder();
+
+  //-------------------------------------------------------------------------
+  /**
+   * Calibrates the ISDA compliant discount curve to the market data.
+   * <p>
+   * This creates the single discount curve for a specified currency.
+   * The curve nodes should be term deposit or fixed-for-Ibor swap, and the number of nodes should be greater than 1.
+   * Typically, the term deposit nodes are used for tenor < 1Y and the swap nodes are used otherwise.
+   * <p>
+   * It is general that the snap date of the market data is different from the valuation date on which 
+   * the resultant curve will be used for pricing CDSs. {@code valuationDate} in {@code marketData} represents 
+   * the snap date, whereas {@code curveValuationDate} does the date on which a credit instrument will be priced.
+   * 
+   * @param curveNode  the curve node
+   * @param curveValuationDate  the curve valuation date
+   * @param curveDcc  the curve day count
+   * @param name  the curve name
+   * @param currency  the currency
+   * @param marketData  the market data
+   * @param computeJacobian  the Jacobian matrices are computed if true
+   * @param refData  the reference data
+   * @return the ISDA compliant discount curve
+   */
+  public IsdaCompliantZeroRateDiscountFactors calibrate(
+      List<IsdaCreditCurveNode> curveNode,
+      LocalDate curveValuationDate,
+      DayCount curveDcc,
+      CurveName name,
+      Currency currency,
+      MarketData marketData,
+      boolean computeJacobian,
+      ReferenceData refData) {
```

Comment:
Add an `IsdaCreditCurveDefinition` class to capture these inputs, (market and reference data would still be passed in here)